plugin/pkg/scheduler/scheduler.go

```diff
@@ -176,6 +190,43 @@ func (sched *Scheduler) schedule(pod *v1.Pod) (string, error) {
 	return host, err
 }
 
+func (sched *Scheduler) preempt(preemptor *v1.Pod, scheduleErr error) (string, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.PodPriority) {
+		glog.V(3).Infof("Pod priority feature is not enabled. No preemption is performed.")
+		return "", nil
+	}
+	preemptor, err := sched.config.PodPreemptor.GetUpdatedPod(preemptor)
```

Comment:
Why do we need this?
Does it give us much value? Even with this there can be race, so I'm not sure what value it introduces.

Reply:
This is needed to get the most recent version of the object. Initially I didn't have this. When it was not there, updating pod annotations after the preemption, was throwing an error that we were trying to update a stale version of the object, because `schedule` updates pod status before calling preempt. We may want to merge the two updates (the one after `schedule` and the one after `preempt`) when preemption is no longer feature gated.